pkg/certificate/signing_requestor.go

```diff
@@ -0,0 +1,487 @@
+	// warningLogInterval defines how often to log warnings for missing callbacks.
+	const warningLogInterval = 12
+
+	secret := obj.(*corev1.Secret)
+
+	v, ok := s.onSignedMap.Load(secret.Name)
+	if !ok {
+		if numRequeues > 0 && numRequeues%warningLogInterval == 0 {
+			logger.Warningf("Received signed Secret for %q with no OnSigned callback registered", secret.Name)
+		}
+
+		return true
+	}
+
+	// Update certificate expiration time for renewal tracking
+	s.updateCertificateExpiration(secret)
+
+	err := v.(OnSignedFn)(secret.Data)
+	if err == nil {
+		return false
+	}
+
+	logger.Errorf(err, "OnSignedFn returned an error for secret %q", secret.Name)
+
+	return true
+}
+
+func (s *signingRequestorImpl) secretName(name string) string {
+	return fmt.Sprintf("%s-%s", name, s.localClusterID)
+}
+
+func deleteIfPresent(ctx context.Context, client dynamic.ResourceInterface, name string) error {
+	err := client.Delete(ctx, name, metav1.DeleteOptions{})
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error deleting Secret %q", name)
+}
+
+// startCertificateRenewalMonitoring starts periodic monitoring of certificate expiration
+func (s *signingRequestorImpl) startCertificateRenewalMonitoring(stopCh <-chan struct{}) {
+	s.renewalTicker = time.NewTicker(CertCheckInterval)
+	s.renewalStopCh = make(chan struct{})
+
+	go func() {
+		defer s.renewalTicker.Stop()
+
+		for {
+			select {
+			case <-s.renewalTicker.C:
+				s.checkCertificateRenewal()
+			case <-s.renewalStopCh:
+				return
+			case <-stopCh:
+				return
+			}
+		}
+	}()
+
+	logger.Info("Started certificate renewal monitoring", "interval", CertCheckInterval)
+}
+
+// stopCertificateRenewalMonitoring stops the certificate renewal monitoring
+func (s *signingRequestorImpl) stopCertificateRenewalMonitoring() {
+	if s.renewalTicker != nil {
+		s.renewalTicker.Stop()
+		s.renewalTicker = nil
+	}
+
+	if s.renewalStopCh != nil {
+		close(s.renewalStopCh)
+		s.renewalStopCh = nil
+	}
+
+	logger.Info("Stopped certificate renewal monitoring")
+}
+
+// updateCertificateExpiration extracts and stores the certificate expiration time
+func (s *signingRequestorImpl) updateCertificateExpiration(secret *corev1.Secret) {
+	certPEM := secret.Data[TLSDataKey]
+	if len(certPEM) == 0 {
+		return
+	}
+
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		logger.Warningf("Failed to decode certificate PEM for secret %q", secret.Name)
+		return
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		logger.Errorf(err, "Failed to parse certificate for secret %q", secret.Name)
+		return
+	}
+
+	// Update the expiration time in our tracking map
+	if v, ok := s.issuedCerts.Load(secret.Name); ok {
+		renewalInfo := v.(certRenewalInfo)
+		renewalInfo.expiresAt = cert.NotAfter
+		s.issuedCerts.Store(secret.Name, renewalInfo)
+
+		logger.V(1).Infof("Updated certificate expiration for %q: expires at %s",
```

Comment:
I think level 1 is the default.

```suggestion
		logger.Infof("Updated certificate expiration for secret %q: expires at %s",
```